lib/jsdom/living/window/SessionHistory.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+const traverseHistory = require("./navigation.js").traverseHistory;
+
+// https://html.spec.whatwg.org/#session-history
+class SessionHistory {
+  constructor(initialEntry, window) {
+    this._window = window;
+    this._historyTraversalQueue = new Set();
+    this.entries = [initialEntry];
+    this.currentIndex = 0;
```

Comment:
I'm trying to encapsulate everything, but using language as close as possible to the spec.  e.g. "Remove all entries after the current entry" is pretty much exactly what it says in the spec in several of the algorithms.  I'll have a go at encapsulating a few more things.

Regarding `currentIndex` vs. `_currentIndex` I'm assuming you would be anti adding something like flow or typescript to this project on account of it requiring a build step?  It would catch this kind of basic mistake, and would have made it much easier to approach the project with the subtle distinction between implementations and idl wrappers actually enforced by some kind of type system. My personal preference would be to gradually convert to typescript, but I'm happy to carry on with plain js assuming that's your preference.